openff/evaluator/backends/dask_kubernetes.py

```diff
@@ -0,0 +1,484 @@
+        # having references to the same object makes things weird
+        worker_container["volumeMounts"] = copy.deepcopy(volume_mounts)
+        scheduler_container["volumeMounts"] = copy.deepcopy(volume_mounts)
+        worker_spec["volumes"] = copy.deepcopy(volumes)
+        scheduler_spec["volumes"] = copy.deepcopy(volumes)
+
+        return spec
+
+    def _generate_volume_specifications(self) -> tuple[list[dict], list[dict]]:
+        """
+        Generate the volume mount and volume specifications for the cluster
+
+        Returns
+        -------
+        tuple[list[dict], list[dict]]
+            A tuple of lists of dictionaries representing the volume mounts
+            and volumes for the cluster, in that order.
+        """
+        volume_mounts = []
+        volumes = []
+
+        for volume in self._volumes + self._secrets:
+            volume_spec = volume._to_volume_spec()
+            volume_mount_spec = volume._to_volume_mount_spec()
+
+            volume_mounts.append(dict(volume_mount_spec))
+            volumes.append(dict(volume_spec))
+
+        return volume_mounts, volumes
+
+    @requires_package("dask_kubernetes")
+    def _generate_worker_spec(self, pod_resources) -> dict[str, dict]:
+        """
+        Generate a Dask Kubernetes worker specification
+        """
+        from dask_kubernetes.operator import make_worker_spec
+
+        resources = pod_resources._to_kubernetes_resource_limits()
+
+        k8s_resources = {
+            "limits": copy.deepcopy(resources),
+            "requests": copy.deepcopy(resources),
+        }
+
+        worker_spec = make_worker_spec(
+            resources=k8s_resources,
+            n_workers=pod_resources._maximum_number_of_workers,
+            image=self._image,
+            env=self._env,
+        )
+
+        # add volume mounts
+        worker_container = worker_spec["spec"]["containers"][0]
+        volume_mounts, volumes = self._generate_volume_specifications()
+        worker_container["volumeMounts"] = copy.deepcopy(volume_mounts)
+        worker_spec["spec"]["volumes"] = copy.deepcopy(volumes)
+
+        # update worker spec
+        if self._resources_per_worker._affinity_specification:
+            worker_spec["spec"]["affinity"] = copy.deepcopy(
+                self._resources_per_worker._affinity_specification
+            )
+
+        # update worker command with resources
+        if self._annotate_resources:
+            pod_resources._update_worker_with_resources(worker_spec["spec"])
+        return worker_spec
+
+    @requires_package("dask_kubernetes")
+    def start(self):
+        from dask_kubernetes.operator import KubeCluster
+        from kubernetes import config
+
+        config.load_kube_config()
+
+        spec = self._generate_cluster_spec()
+        self._cluster = KubeCluster(
+            namespace=self._namespace, custom_cluster_spec=spec, **self._cluster_kwargs
+        )
```

Comment:
To do: a PVC would need to be started already for the connected Dask cluster to work. This can be done via the Kubernetes API, although I currently do it separately to the DaskKubernetesBackend. Should it be migrated into this class to autostart the pvc?